src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoDeserializer.cs

```diff
@@ -310,6 +317,142 @@ private void deserializeObjectInto<T>(
             PocoDeserializationHelper.RunInstanceValidation(target, Settings.Validator, context, state.Errors);
         }
     }
+    
+    private void deserializeUnknownPropertiesInto<T>(
+        T target,
+        string propertyName,
+        ref Utf8JsonReader reader,
+        FhirJsonPocoDeserializerState state,
+        bool stayOnLastToken = false) where T : Base
+    {
+        target.TryGetValue(propertyName, out var value);
+
+        var (name, choiceType) = tryDetectChoiceTypeFromName(propertyName);
+
+        // move past property name
+        reader.Read();
+
+        object? result;
+        if(choiceType is not null)
+        {
+            state.Path.EnterElement(name, null, choiceType.IsPrimitive);
+            
+            result = DeserializeFhirPrimitive(value as PrimitiveType, name, choiceType, null, ref reader, new(), state);
+            
+            state.Path.ExitElement();
+        }
+        else
+        {
+            state.Path.EnterElement(name, reader.TokenType == JsonTokenType.StartArray ? 0 : null, isJsonPrimitive(ref reader));
+            
+            result = deserializeUnexpectedJsonValue(propertyName, ref reader, state, stayOnLastToken);
+            
+            state.Path.ExitElement();
+        }
+        
+        target.SetValue(name, result);
+        (string name, ClassMapping? choiceType) tryDetectChoiceTypeFromName(string propertyName)
+        {
+            var span = propertyName.AsSpan();
+            for(int i = 0; i < span.Length; i++)
+            {
+                if (!char.IsUpper(span[i])) 
+                    continue;
+
+                var subSpan = span.Slice(i);
+                if (subSpan.IsEmpty)
+                    break;
+                
+                var choiceMapping = _inspector.FindClassMapping(subSpan.ToString());
+                if (choiceMapping is not null)
+                    return (span[..i].ToString(), choiceMapping);
+            }
+            return (propertyName, null);
+        }
+    }
+
+    private object? deserializeUnexpectedJsonValue(string propertyName, ref Utf8JsonReader reader, FhirJsonPocoDeserializerState state, bool stayOnLastToken, ClassMapping? propertySuggestion = null)
```

Comment:
Nice, I think this is going to be the main workhorse, in fact we could call this straight from the main TryDeserializeResouirce and TryDeserializeObject functions - we're only *expecting* something, but we're no longer bound to it, so this dynamic function is really nice.